lib_stusb_impl/usbd_impl.c

```diff
@@ -1018,8 +1018,13 @@ uint8_t USBD_HID_DataIn_impl(USBD_HandleTypeDef *pdev, uint8_t epnum)
 
 uint8_t USBD_HID_DataOut_impl(USBD_HandleTypeDef *pdev,
                               uint8_t             epnum,
-                              uint8_t            *buffer,
-                              apdu_buffer_t      *apdu_buf)
+#ifndef HAVE_USB_HIDKBD
+                              uint8_t       *buffer,
+                              apdu_buffer_t *apdu_buf)
+#else
+                              __attribute__((unused)) uint8_t       *buffer,
+                              __attribute__((unused)) apdu_buffer_t *apdu_buf)
+#endif  // HAVE_USB_HIDKBD

```

Comment:
Maybe this to confuse less IDE?
```suggestion
#ifndef HAVE_USB_HIDKBD
                              uint8_t       *buffer,
                              apdu_buffer_t *apdu_buf
#else
                              __attribute__((unused)) uint8_t       *buffer,
                              __attribute__((unused)) apdu_buffer_t *apdu_buf
#endif  // HAVE_USB_HIDKBD
                             )
```

Reply:
I'll update, but I bet the linter will whine about something.